packages/www/pages/transact/eth/index.tsx

```diff
@@ -0,0 +1,454 @@
+      return;
+    }
+  }
+
+  try {
+    await ethereum.request({
+      method: "wallet_addEthereumChain",
+      params: [networks[chainId].spec],
+    });
+    logger("Successfully added Polygon network to MetaMask.");
+    return;
+  } catch (err) {
+    logger(`Error adding Polygon network: ${err.message}`);
+  }
+}
+
+const TransactEth = () => {
+  const { status, connect, account, chainId, ethereum } = useMetaMask();
+  const isMinting = useToggleState();
+  const web3 = useMemo(() => new Web3(ethereum), [ethereum]);
+
+  const initState = useMemo(() => {
+    if (typeof window === "undefined") {
+      return {};
+    }
+    const searchParams = new URLSearchParams(window.location.search);
+    return {
+      contractAddress: searchParams.get("contractAddress"),
+      tokenUri: searchParams.get("tokenUri"),
+      recipient: searchParams.get("recipient"),
+    };
+  }, [typeof window !== "undefined" && window?.location?.search]);
+  const defaultContractAddress = useMemo<string>(
+    () => networks[chainId]?.defaultContract,
+    [chainId]
+  );
+  const [state, setState] = useState(initState);
+  type State = typeof state;
+  const setStateProp = <T extends keyof State>(prop: T, value: State[T]) => {
+    setState({ ...state, [prop]: value });
+  };
+
+  const [logs, setLogs] = useState<JSX.Element[]>([]);
+  const addLog = useCallback(
+    (log: JSX.Element | string) =>
+      setLogs((prev) => [
+        ...prev,
+        <>
+          [{new Date().toLocaleTimeString()}] {log}
+        </>,
+      ]),
+    [setLogs]
+  );
+
+  const onClickMint = useCallback(async () => {
+    isMinting.onOn();
+    try {
+      await mintNft(
+        web3,
+        state.contractAddress ?? defaultContractAddress,
+        account,
+        state.recipient ?? account,
+        state.tokenUri,
+        addLog
+      );
+    } finally {
+      isMinting.onOff();
+    }
+  }, [state, web3, defaultContractAddress, account, addLog]);
+
+  const onClickSwitchNetwork = useCallback(() => {
+    setLogs([]);
+    return switchNetwork(ethereum, defaultNet.spec.chainId, addLog);
+  }, [setLogs, ethereum, addLog]);
+
+  const onClickConnect = useCallback(() => {
+    setLogs([]);
+    return connect();
+  }, [setLogs, connect]);
+
+  return (
+    <Layout {...Content.metaData}>
+      <Guides backgroundColor="$mauve2" />
+      <Box css={{ position: "relative" }}>
+        <Container
+          size="3"
+          css={{
+            px: "$6",
+            py: "$7",
+            width: "100%",
+            "@bp3": {
+              py: "$8",
+              px: "$4",
+            },
+          }}>
+          <Flex
+            css={{
+              alignItems: "center",
+              justifyContent: "center",
+              flexGrow: 1,
+              flexDirection: "column",
+            }}>
+            <AlertDialog open={true}>
+              <AlertDialogContent
+                css={{ maxWidth: 450, px: "$5", pt: "$4", pb: "$4" }}
```

Comment:
Instead of having the header and the footer with an empty page on the background, could this dialog cast a box shadow? Or is there something already in place for this kind of standalone dialogs? 
```
boxShadow: "0 0 100vw 100vw rgb(200 200 200)"
```

Although I guess it will be a separate full page and / or a dialog in the correct place (e.g. Asset Table dashboard) and never a standalone dialog?

Reply:
Yeah my idea is that soon we will extract a dialog component from this and inject it in some other pages like asset. Didn't want to spend tooo much time on the styling there, so the one thing I'm gonna try to do is at least to move the footer down in the page, so it doesn't look that weird